unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuildModeController.cs

```diff
@@ -0,0 +1,1069 @@
+using Builder;
+using Builder.Gizmos;
+using DCL;
+using DCL.Components;
+using DCL.Configuration;
+using DCL.Controllers;
+using DCL.Helpers;
+using DCL.Models;
+using DCL.Tutorial;
+using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.XR;
+
+public class BuildModeController : ViewController
```

Comment:
Right now this is the "brain" that connects all the functionality across the Builder In World system. We can split it but it will create a more complex system and it will be a big change. However, it will improve the readability, maintainability, and scalability.

Let's see what others think about that!

Reply:
I missed this script because github wouldn't show the diff implicitly due to the amount of lines haha. I agree with Santi, this is too overkill and quite hard to review.

Reply:
Ok! Let's move this as a rework of the script